### PR TITLE
[Feature] Add SROIE Converter

### PR DESCRIPTION
**Pull request by xinke-wang**
## Motivation

Add data converter for [SROIE](https://rrc.cvc.uab.es/?ch=13) dataset

## Modification

add tools for loading and converting labels from txt annotations

- tools/data/textdet/sroie_converter.py
- tools/data/textrecog/sroie_converter.py

update document for SROIE dataset

- docs/en/datasets/det.md
- docs/en/datasets/recog.md

## Use cases (Optional)

Screenshot from 2022-03-03 23-18-00

### CODE REVIEW

**Review by gaotongxiao — COMMENTED** [review]
SROIE's test set is also publicly available now. We should at least mention that in the documentation and provide an option in conversion scripts for not splitting the dataset.

**Review by gaotongxiao — COMMENTED** [review]
This dataset also reminds me of MMOCR's poor support of space recognition. We need to implement this feature ASAP.

**Comment by xinke-wang** [comment]
> SROIE's test set is also publicly available now. We should at least mention that in the documentation and provide an option in conversion scripts for not splitting the dataset.

Agree. I replace the VAL split with the official TEST split. So it keeps consistency with the existing converters.

File: docs/en/datasets/det.md [hunk 1]
```diff
@@ -212,3 +218,29 @@ rm dataset.zip && rm -rf dataset
...
+### SROIE
+
+- Step1: Download `0325updated.task1train(626p).zip`, `task1&2_test(361p)-20220304T064017Z-001`, and `text.task1&2-test（361p).zip` from [homepage](https://rrc.cvc.uab.es/?ch=13&com=downloads) to `sroie/`
+- Step2:
+
+```bash
+mkdir sroie && cd sroie
+mkdir imgs && mkdir annotations && mkdir imgs/training
+
+unzip -q 0325updated.task1train\(626p\).zip && unzip -q task1\&2_test\(361p\)-20220304T064017Z-001.zip && unzip -q text.task1\&2-test（361p\).zip
```
Comment (gaotongxiao) [inline docs/en/datasets/det.md:231]:
```suggestion
unzip -q 0325updated.task1train\(626p\).zip && unzip -q task1\&2_test\(361p\).zip && unzip -q text.task1\&2-test（361p\).zip
```
The timestamp is created by GDrive! Might need to *note* it down.

File: tools/data/textdet/sroie_converter.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    Args:
+        gt_file (list): The list of tuples (image_file, groundtruth_file)
+        img_info (int): The dict of the img and annotation information
+
+    Returns:
+        img_info (list): The dict of the img and annotation information
+    """
+
+    with open(gt_file, 'r', encoding='unicode_escape') as f:
```
Comment (gaotongxiao) [inline tools/data/textdet/sroie_converter.py:110]: why `unicode_escape`

File: docs/en/datasets/det.md (hunk 1 above)
Comment (xinke-wang) [reply docs/en/datasets/det.md:231]: Forgot to delete it.

File: docs/en/datasets/det.md [hunk 3]
```diff
@@ -212,3 +218,29 @@ rm dataset.zip && rm -rf dataset
 ```bash
 python tools/data/textdet/funsd_converter.py PATH/TO/funsd --nproc 4
 ```
+
+### SROIE
+
+- Step1: Download `0325updated.task1train(626p).zip`, `task1&2_test(361p)-20220304T064017Z-001`, and `text.task1&2-test（361p).zip` from [homepage](https://rrc.cvc.uab.es/?ch=13&com=downloads) to `sroie/`
```
Comment (gaotongxiao) [inline docs/en/datasets/det.md:224]: delete timestamp

File: docs/en/datasets/det.md [hunk 4]
```diff
@@ -212,3 +218,29 @@ rm dataset.zip && rm -rf dataset
...
+
+unzip -q 0325updated.task1train\(626p\).zip && unzip -q task1\&2_test\(361p\).zip && unzip -q text.task1\&2-test（361p\).zip
+
+# For images
+mv 0325updated.task1train\(626p\)/*.jpg imgs/training && mv task1_2_test\(361p\)/ imgs/test
+
+# For annotations
+mv 0325updated.task1train\(626p\) annotations/training && mv text.task1_2-test（361p\)/ annotations/test
+
+rm 0325updated.task1train\(626p\).zip && rm task1\&2_test\(361p\)-20220304T064017Z-001.zip && rm text.task1\&2-test（361p\).zip
```
Comment (gaotongxiao) [inline docs/en/datasets/det.md:239]: delete timestamp

File: docs/en/datasets/recog.md [hunk 5]
```diff
@@ -321,3 +327,29 @@ rm dataset.zip && rm -rf dataset
 ```bash
 python tools/data/textrecog/funsd_converter.py PATH/TO/funsd --nproc 4
 ```
+
+### SROIE
+
+- Step1: Step1: Download `0325updated.task1train(626p).zip`, `task1&2_test(361p)-20220304T064017Z-001`, and `text.task1&2-test（361p).zip` from [homepage](https://rrc.cvc.uab.es/?ch=13&com=downloads) to `sroie/`
```
Comment (gaotongxiao) [inline docs/en/datasets/recog.md:333]: delete timestamp

File: docs/en/datasets/recog.md [hunk 6]
```diff
@@ -321,3 +327,29 @@ rm dataset.zip && rm -rf dataset
...
+- Step2:
+
+```bash
+mkdir sroie && cd sroie
+mkdir imgs && mkdir annotations && mkdir imgs/training
+
+unzip -q 0325updated.task1train\(626p\).zip && unzip -q task1\&2_test\(361p\)-20220304T064017Z-001.zip && unzip -q text.task1\&2-test（361p\).zip
```
Comment (gaotongxiao) [inline docs/en/datasets/recog.md:340]: delete timestamp

File: docs/en/datasets/recog.md [hunk 7]
```diff
@@ -321,3 +327,29 @@ rm dataset.zip && rm -rf dataset
...
+
+# For images
+mv 0325updated.task1train\(626p\)/*.jpg imgs/training && mv task1_2_test\(361p\)/ imgs/test
+
+# For annotations
+mv 0325updated.task1train\(626p\) annotations/training && mv text.task1_2-test（361p\)/ annotations/test
+
+rm 0325updated.task1train\(626p\).zip && rm task1\&2_test\(361p\)-20220304T064017Z-001.zip && rm text.task1\&2-test（361p\).zip
```
Comment (gaotongxiao) [inline docs/en/datasets/recog.md:348]: delete timestamp

**Review by gaotongxiao — APPROVED** [review]

**Review by Mountchicken — APPROVED** [review]
Tested on sproject, everything works fine.

**Pull request merged**